### DISCUSSION TITLE
400: Not a JSON request

### DISCUSSION THREAD

**Discussion by AnssiAhola** [joalla/discogs_client#17] [Support] [answered]
Hi.

I was trying to implement a feature where you can add new listings to inventory/marketplace,
but I get '400: Not a JSON request' as a response.
I have authenticated with oauth and have filled out all the required seller settings in discogs.

Implementation:
https://github.com/AnssiAhola/discogs_client/blob/4ff826e977e20d0d7809fc404a33624b9ffc212e/discogs_client/models.py#L401-L421

I was calling it like this
`me.inventory.add_listing(1902020, condition='Near Mint (NM or M-)', price='999' )`

Edit:
Also returns '400' when trying to save changes on APIObject like so:
```
item = me.wantlist[0]
item.notes = 'New note - via API'
item.save()
```

The API documentation says that the request headers must have 'Content-Type: application/json',
so i tried to pass a boolean down the line to the fetcher and if true it would replace the header with json,
but this resulted in a ValueError from oauthlib, so maybe not the issue then.

Any ideas what might cause this. Maybe it's something at the Discogs end and I'm barking at the wrong tree?

Thought this would fit better in here instead Issues. Or Forums?

PS.
Also as a side note I saw this in the fetchers
https://github.com/joalla/discogs_client/blob/857dc0475786cc7f289ea8534ff454d7dbb8552c/discogs_client/fetchers.py#L96-L101
Why is the result of the line 97 basically ignored?

**Comment by AnssiAhola** [comment, marked as answer]
#18 fixes this.

**Reply by alifhughes** [reply]
Hey @AnssiAhola ! Sorry for the delay in response, I'll take a look at this over the next 24 hours. Thanks for opening a PR to fix it too 🙂 I'll take a look
